app/models/user.rb

```diff
@@ -3,4 +3,25 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable, :confirmable
+  has_many :orders
+  attr_accessor :stripe_card_token
+
+  def cart
+    orders.where(completed: false).first_or_create
```

Comment:
We don't need the `completed` column, you can just use the check for a stripe token instead.